NDTensors/src/Unwrap/test/runtests.jl

```diff
@@ -0,0 +1,90 @@
+using Test
+using NDTensors.Unwrap
+using NDTensors
+using LinearAlgebra
+
+include("../../../test/device_list.jl")
+@testset "Testing Unwrap" for dev in devices_list(ARGS)
+  v = dev(Vector{Float64}(undef, 10))
+  vt = transpose(v)
+  va = v'
+
+  E = expose(v)
+  Et = expose(vt)
+  Ea = expose(va)
+  v_type = typeof(v)
+  e_type = eltype(v)
+  @test typeof(E) == Exposed{v_type,v_type}
+  @test typeof(Et) == Exposed{v_type,LinearAlgebra.Transpose{e_type,v_type}}
+  @test typeof(Ea) == Exposed{v_type,LinearAlgebra.Adjoint{e_type,v_type}}
+
+  @test parent(E) == v
+  @test parent(Et) == v
+  @test parent(Ea) == v
+  @test transpose(E) == vt
+  @test cpu(E) == v
+  @test cpu(Et) == vt
+
+  m = reshape(v, (5, 2))
+  mt = transpose(m)
+  ma = m'
+  E = expose(m)
+  Et = expose(mt)
+  Ea = expose(ma)
+
+  m_type = typeof(m)
+  @test typeof(E) == Exposed{m_type,m_type}
+  @test typeof(Et) == Exposed{m_type,LinearAlgebra.Transpose{e_type,m_type}}
+  @test typeof(Ea) == Exposed{m_type,LinearAlgebra.Adjoint{e_type,m_type}}
+
+  o = dev(Vector{Float32})(undef, 1)
+  expose(o)[] = 2
+  expose(o)[] == 2
+
+  fill!(m, 0)
+  @test any(!Base.isinf, expose(m))
+
+  mp = copy(Ea)
+  mp == ma
+  fill!(ma, 2.0)
+  copyto!(expose(mp), expose(ma))
+  mp == ma
+
+  q, r = qr(expose(mp))
+  @test q * r ≈ mp
+
+  q, r = Unwrap.qr_positive(expose(mp))
+  @test q * r ≈ mp
+
+  square = NDTensors.cu(rand(Float64, (10, 10)))
+  square = (square + square) ./ 2.0
+  ## CUDA only supports Hermitian or Symmetric eigen decompositions
+  ## So I symmetrize square and call symetric here
+  ## TODO I am not sure how to check this because square * U[:,1] != l[1] * U[:,1]
+  l, U = eigen(expose(Symmetric(square)))
```

Comment:
Does it work to test:
```julia
@test Symmetric(square) * U ≈ U * Diagonal(l)
```
?

Reply:
Ahh I realize I symmetrized square wrong so if I fix this the test `square * U \approx U * Diagonal(l)` works regardless of the `Symmetric` wrapper.